### PR TITLE
Create account page

### PR DESCRIPTION
**Pull request by jaypark25**
Changes
- Made /signup page to create user accounts using React-hook-form & Zod
 
Issue:
#31 

Out Of Scope
Connecting to backend to make the actual account 

Notes
- Made password check through Zod's refine
- Uhh for InternationalStudent & BizTech in the past question, currently returns a string instead of a boolean because I couldn't pass in a string -- need to deal with this before submissions
- I basically made components for all the renders because I didn't want to repeat styling. Let me know if this is problematic at all.
- Not much to make responsible except text sizes lol
(Web)
image
image

(Mobile)
image
image

### CODE REVIEW

**Comment by jaypark25** [comment]
how do i fix 1000 lines of package-lock conflict help @voctory

**Comment by voctory** [comment]
> how do i fix 1000 lines of package-lock conflict help @voctory

@jaypark25 Once you fix the merge conflict for `package.json`, you can run `npm install` which automatically generates a new `package-lock.json`. After, you should be fine to overwrite the `package-lock.json` with your new copy

https://stackoverflow.com/questions/54124033/deleting-package-lock-json-to-resolve-conflicts-quickly

File: src/components/SignupForm/SignUpForm.tsx [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    First = "1st Year",
+    Second = "2nd Year",
+    Third = "3rd Year",
+    Fourth = "4th Year",
+    Fifth = "5th Year",
+    Other = "Other",
+    NotApplicable = "Not Applicable"
+}
+
+enum dietaryRestrictions {
```
Comment (AllanT102) [inline src/components/SignupForm/SignUpForm.tsx:42]: capitalize

**Comment by jaypark25** [comment]
Oh I actually read that exact stack overflow page lol @voctory but I just read it could be really bad so I didn't do it i assume it's okay because our versions aren't too diverged yet

**Comment by AllanT102** [comment]
@jaypark25 Sorry, seems like the figma design was missing the different views depending on what the initial option was selected. Play around with the "Please select the option that's most relevant to you " and please implement the other questions. They should be dynamic. 

Here is how the data would be passed to the backend/payments https://github.com/ubc-biztech/bt-web/blob/f99602b602bf96178dddd54b8294deed962ad2e5/src/pages/public/RegistrationForms/OAuthUserMembershipForm/index.js This registration will basically create a "User" object in the users table. I know there are extra fields that are in the "paymentBody" const in the file – they're needed and we're victor and i are trying to figure it out how to handle that rn. We'll keep you updated, but please work on these for now

File: public/assets/icons/back_icon.svg
Comment (kevinxiao27) [inline public/assets/icons/back_icon.svg:1]: nit: Undo2 from lucide-react looks pretty similar, would have to apply some tailwind classes to flip/colour

Comment (kevinxiao27) [reply public/assets/icons/back_icon.svg:1]: just noticed PR still in progress will comeback whendone

**Comment by jaypark25** [comment]
implemented the dynamic form -- think we should test @ sync lol
not sure if you want me to do the registriation part as well because I know benny doing the signup workflow? let me know and I can take a look @AllanT102

**Comment by AllanT102** [comment]
@jaypark25 Can you implement the API call that navigates user to stripe? https://dev.app.ubcbiztech.com/signup basically the page ur creating is the "User + Membership" one. I'll send you code snippet.

**Pull request closed without merging**